File: Tech-Module/Programming_Fundametals/08_Dictionaries_Lambda_Expressions_And_LINQ/Exercises/01_Phonebook/Phonebook.cs
namespace _01.Phonebook
{
    using System;
    using System.Collections.Generic;

    public class Phonebook
    {
        public static void Main()
        {
            var dictionary = new Dictionary<string, string>();
            var input = Console.ReadLine();

            while (input != "END")
            {
                var tokens = input.Split(' ');
                var command = tokens[0];

                if (command == "A")
                {
                    var newName = tokens[1];
                    var newPhone = tokens[2];
                    dictionary[newName] = newPhone;
                }

                if (command == "S")
                {
                    var name = tokens[1];

                    if (dictionary.ContainsKey(name))
                    {
                        Console.WriteLine($"{name} -> {dictionary[name]}");
                    }
                    else
                    {
                        Console.WriteLine($"Contact {name} does not exist.");
                    }
                }

                input = Console.ReadLine();
            }
        }
    }
}

File: Tech-Module/Software_Technologies/09. Java-Syntax-And-Basics/src/exercises/Phonebook.java
package exercises;

import java.util.*;

public class Phonebook {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] input = scanner.nextLine().split(" ");
        Map<String, String> phonebook = new HashMap<>();

        while (!(input[0].equals("END"))) {
            if(input[0].equals("A")) {
                String name = input[1];
                String phone = input[2];

                phonebook.putIfAbsent(name, "");
                phonebook.put(name, phone);
            }

            if (input[0].equals("S")) {
                String name = input[1];

                int count = 0;

                for (String key : phonebook.keySet()) {
                    if(key.equals(name)) {
                        System.out.println(key + " -> " + phonebook.get(key));
                        count++;
                    }
                }

                if(count == 0) {
                    System.out.printf("Contact %s does not exist.\n", name);
                }
            }

            input = scanner.nextLine().split(" ");
        }
    }
}
